src/ZeroLog/JsonWriter.cs

```diff
@@ -0,0 +1,319 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text.Formatting;
+
+namespace ZeroLog
+{
+    public static unsafe class JsonWriter
+    {
+        public static void WriteJsonToStringBuffer(StringBuffer stringBuffer, IList<IntPtr> keyValuePointers,
+                                                   string[] strings)
+        {
+            stringBuffer.Append(LogManager.Config.JsonSeparator);
+            stringBuffer.Append("{ ");
+            for (var i = 0; i < keyValuePointers.Count; i++)
+            {
+                var argPointer = (byte*)keyValuePointers[i].ToPointer();
+                var dataPointer = argPointer;
+
+                // Key.
+                AppendJsonValue(stringBuffer, strings, ref dataPointer);
+
+                stringBuffer.Append(": ");
+
+                // Value.
+                AppendJsonValue(stringBuffer, strings, ref dataPointer);
+
+                if (i != keyValuePointers.Count - 1)
+                    stringBuffer.Append(", ");
+            }
+
+            stringBuffer.Append(" }");
+        }
+
+        private static void AppendJsonValue(StringBuffer stringBuffer, IReadOnlyList<string> strings, ref byte* dataPointer)
+        {
+            var argument = *dataPointer;
+            dataPointer += sizeof(ArgumentType);
+
+            var argumentType = (ArgumentType)(argument & ArgumentTypeMask.ArgumentType);
+
+            switch (argumentType)
+            {
+                case ArgumentType.KeyString:
+                case ArgumentType.String:
+                    stringBuffer.Append('"');
+                    if (NeedsEscaping(strings[*dataPointer]))
```

Comment:
It would probably be faster to always escape and append and skip `NeedsEscaping` entirely, since `StringBuffer` simply appends char-by-char anyway.